crates/test-utils/src/transaction.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::authority::get_client;
+use crate::messages::{create_publish_move_package_transaction, make_certificates};
+use std::collections::HashMap;
+use std::path::PathBuf;
+use sui_config::ValidatorInfo;
+use sui_core::authority_client::AuthorityAPI;
+use sui_types::base_types::ObjectRef;
+use sui_types::error::SuiResult;
+use sui_types::messages::{
+    ConfirmationTransaction, ConsensusTransaction, Transaction, TransactionEffects,
+    TransactionInfoResponse,
+};
+use sui_types::object::{Object, Owner};
+
+pub async fn publish_package(
+    gas_object: Object,
+    path: PathBuf,
+    configs: &[ValidatorInfo],
+) -> ObjectRef {
+    let transaction = create_publish_move_package_transaction(gas_object, path);
+    let effects = submit_single_owner_transaction(transaction, configs).await;
+    parse_package_ref(&effects).unwrap()
+}
+
+/// Helper function to publish the move package of a simple shared counter.
+pub async fn publish_counter_package(gas_object: Object, configs: &[ValidatorInfo]) -> ObjectRef {
+    let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    path.push("../../sui_programmability/examples/basics");
```

Comment:
Pre-existing, but in general I'm not a fan of depending on the example code in tests because it tends to be brittle...